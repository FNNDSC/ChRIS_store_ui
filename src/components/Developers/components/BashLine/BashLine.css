.bash-line-container {
  /* display */
  display: flex;

  /* position */
  position: relative;
}

.bash-line-command {
  /* color */
  background: #ededed;
  background: var(--pf-black-200);

  /* display */
  width: 0;
  height: 48px;
  padding: 10px;
  padding-right: 46px;

  /* font */
  font-family: 'Inconsolata';
  font-size: medium;

  /* border */
  border-left-color: #00659c;
  border-left: 0.5em solid var(--pf-blue-500);

  /* other */
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  /* flex */
  flex: 1;
}

/* import iconsolata font */
@font-face {
  font-family: 'Inconsolata';
  src: url('../../../../assets/fonts/Inconsolata/Inconsolata-Regular.ttf') format('ttf'),
       url('../../../../assets/fonts/Inconsolata/Inconsolata-Regular.woff2') format('woff2'),
       url('../../../../assets/fonts/Inconsolata/Inconsolata-Regular.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}
