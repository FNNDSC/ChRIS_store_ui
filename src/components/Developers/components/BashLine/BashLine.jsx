import React from 'react';
import { Button, clipboardCopyFunc, Tooltip } from '@patternfly/react-core';
import { CopyIcon } from '@patternfly/react-icons';
import PropTypes from 'prop-types';
import styles from './BashLine.module.css';

function BashLine({ command }) {
  return (
    <div className={styles.bashLineContainer}>
      <div className={styles.bashLineCommand}>
        {`$ ${command}`}
      </div>
      <Tooltip content="Copy">
        <Button
          isSmall
          variant="control"
          aria-label="Copy"
          onClick={(e) => clipboardCopyFunc(e, command)}
        >
          <CopyIcon />
        </Button>
      </Tooltip>
    </div>
  );
}

BashLine.propTypes = { command: PropTypes.string.isRequired };

export default BashLine;
