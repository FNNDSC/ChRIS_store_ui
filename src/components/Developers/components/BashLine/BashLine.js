import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CopyToClipboard from '../CopyToClipboard/CopyToClipboard';
import './BashLine.css';

class BashLine extends Component {
  constructor() {
    super();
    this.tooltipShown = false;
  }

  render() {
    return (
      <div className="bash-line-container" {...this.props}>
        <div className="bash-line-command">
          {`$ ${this.props.command}`}
        </div>
        <CopyToClipboard clipboardText={this.props.command} />
      </div>
    );
  }
}

BashLine.propTypes = { command: PropTypes.string.isRequired };

export default BashLine;
