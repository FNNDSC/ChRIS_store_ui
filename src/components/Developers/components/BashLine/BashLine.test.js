import React from 'react';
import { mount, shallow } from 'enzyme';
import BashLine from './BashLine';

describe('BashLine', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<BashLine command="" />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mounted BashLine', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<BashLine command="" />);
  });

  it('renders the value of command', () => {
    wrapper.setProps({ command: 'command' });
    expect(wrapper.find('.bash-line-command').text()).toEqual('$ command');
  });
});
