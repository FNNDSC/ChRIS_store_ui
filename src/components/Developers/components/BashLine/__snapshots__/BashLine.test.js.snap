// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BashLine should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BashLine
    command=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="bash-line-command"
>
          $ 
</div>,
        <CopyToClipboard
          clipboardText=""
/>,
      ],
      "className": "bash-line-container",
      "command": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "$ ",
          "className": "bash-line-command",
        },
        "ref": null,
        "rendered": "$ ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "clipboardText": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="bash-line-command"
>
            $ 
</div>,
          <CopyToClipboard
            clipboardText=""
/>,
        ],
        "className": "bash-line-container",
        "command": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "$ ",
            "className": "bash-line-command",
          },
          "ref": null,
          "rendered": "$ ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "clipboardText": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
